static/bonescript.js

```diff
@@ -24,12 +15,24 @@ function setTargetAddress(address, handlers) {
         script.type = 'text/javascript';
         script.src = url;
         script.charset = 'UTF-8';
+        script.onerror = function() {
```

Comment:
Why are you putting this in the HTML document context? The local namespace is perfectly fine to use without reaching into the HTML document.

Glad you figured out that your synchronous logic wouldn't work, but my preference is that you create a branch with your changes 'squashed' into logic chunks for my consumption. Sure, I don't always do the same myself, but if I'm submitting my code to someone else, I'll make sure the individual commits are readable and functional.